### PR TITLE
Complete saveat for GPUTsit5

### PR DESCRIPTION
**Pull request by utkarsh530**
MWE:

```
using DiffEqGPU, SimpleDiffEq, StaticArrays, CUDA, OrdinaryDiffEq
using Plots

CUDA.allowscalar(false)

function f(u, p, t)
    du1 = -u[1]
    return SVector{1}(du1)
end

u0 = @SVector [10.0f0]
prob = ODEProblem{false}(f, u0, (0.0f0, 10.0f0))
prob_func = (prob, i, repeat) -> remake(prob, p = prob.p)
monteprob = EnsembleProblem(prob, safetycopy = false)

sol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(), trajectories = 2,
                    adaptive = false, dt = 0.1f-1, abstol = 1.0f-8, reltol = 1.0f-5,saveat = [0.0f0,4.0f0])
```

```
julia> sol[1].u
2-element view(::Matrix{SVector{1, Float32}}, :, 1) with eltype SVector{1, Float32}:
 [10.0]
 [0.1831558]
```

@ChrisRackauckas see the claim that I was talking about.

### CODE REVIEW

File: test/gpu_tsit5_tests.jl [hunk 1]
```diff
@@ -33,6 +33,15 @@ bench_asol = solve(prob, Tsit5(), dt = 0.1f-1, save_everystep = false, abstol =
...
 @show norm(bench_asol.u[end] - asol[1].u[end]) < 1e-4
 
+### solve parameters
+
+sol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(), trajectories = 2,
+            adaptive = false, dt = 0.1f0, saveat = [0.0f0, 4.0f0])
+
+asol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(), trajectories = 2,
+             adaptive = true, dt = 0.1f-1, abstol = 1.0f-8, reltol = 1.0f-5,
+             saveat = [0.0f0, 4.0f0])
```
Comment (ChrisRackauckas) [inline test/gpu_tsit5_tests.jl:43]: test that the values and lengths are as expected.

**Comment by ChrisRackauckas** [comment]
Looks like there's real test failures?

**Comment by utkarsh530** [comment]
Isn't it in the LinearSolve.jl: https://buildkite.com/julialang/diffeqgpu-dot-jl/builds/235#01833d7b-971a-4899-b9a6-89c11ac55f9d/519-860 ?

**Comment by ChrisRackauckas** [comment]
Why is p nothing?

https://github.com/SciML/DiffEqGPU.jl/blob/master/src/DiffEqGPU.jl#L765

Master was passing so this should get figured out.

**Comment by ChrisRackauckas** [comment]
Seems like a test failure in GPUTsit5 https://buildkite.com/julialang/diffeqgpu-dot-jl/builds/243#018340b0-cd62-4c93-9b7b-3c9ce9e69526/357-861

File: test/gpu_tsit5_tests.jl [hunk 2]
```diff
@@ -26,12 +26,42 @@ asol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(), trajectories = 10,
...
 
-bench_sol = solve(prob, Tsit5(), adaptive = false, dt = 0.1f0)
+bench_sol = solve(prob, Tsit5(), adaptive = false, dt = 0.01f0)
 bench_asol = solve(prob, Tsit5(), dt = 0.1f-1, save_everystep = false, abstol = 1e-8,
                    reltol = 1e-5)
 
-@show norm(bench_sol.u[end] - sol[1].u[end]) < 2e-2
-@show norm(bench_asol.u[end] - asol[1].u[end]) < 1e-4
+@test norm(bench_sol.u[end] - sol[1].u[end]) < 5e-3
+@test norm(bench_asol.u - asol[1].u) < 1e-4
```
Comment (ChrisRackauckas) [inline test/gpu_tsit5_tests.jl:34]:
```suggestion
@test norm(bench_asol.u - asol[1].u) < 4e-4
```

**Comment by ChrisRackauckas** [comment]
Run the formatter and this should be good to go.

**Comment by utkarsh530** [comment]
Thanks for the help! I see what was the issue.
Can you merge it? I'll rebase the master then with #174.

**Pull request merged**